creusot/src/backend/logic.rs

```diff
@@ -199,6 +204,49 @@ fn body_module<'tcx>(ctx: &mut Why3Generator<'tcx>, def_id: DefId) -> (Module, C
     (Module { name, decls }, summary)
 }
 
+fn subst_qname(body: &mut Exp, name: &Ident, lim_name: &Ident) {
+    struct QNameSubst<'a>(&'a Ident, &'a Ident);
+
+    impl<'a> ExpMutVisitor for QNameSubst<'a> {
+        fn visit_mut(&mut self, exp: &mut Exp) {
+            match exp {
+                Exp::QVar(qname, _) if qname.module.is_empty() && &qname.name == self.0 => {
+                    *exp = Exp::pure_var(self.1.clone())
+                }
+                _ => super_visit_mut(self, exp),
+            }
+        }
+    }
+
+    QNameSubst(name, lim_name).visit_mut(body)
+}
```

Comment:
This does seem a little sketch, I'm not sure what a better way to do this would be? Would substituting the Pearlite term work?

Reply:
Do you mean that I should move this transformation to the `pearlite::Term` level instead of using `why3::Exp`? That would probably make things more complicated in other ways.

Reply:
I would be curious to know what the complexity is, I want to avoid doing work at the why3 level as much as possible, especially if this is meant to be portable across backends (which I'll start work on a second soon).

Reply:
Function calls in `pearlite::Term` uses `DefId`s for function calls I'm not sure how I could do this other than by generating the limited function at the macro level (having `logic` `ghost` and `predicate` generate an addition Rust function that is used as the limited function)

Reply:
Hmm fair. We'll need to figure out a better solution in the future